Snake.js

```diff
@@ -199,32 +196,27 @@ Snake.prototype.createGeometry = function( geometry  , numOf ){
       normals[ index * 3  + 6 ] = n3.x;
       normals[ index * 3  + 7 ] = n3.y;
       normals[ index * 3  + 8 ] = n3.z;
-    
+   
+    } 
+        
+  for( var i = 0; i < numOf; i++ ){
       var y = (Math.floor( i / lookupSize ))/ lookupSize;
       var x = (i - ( (Math.floor( i / lookupSize )) * lookupSize )) / lookupSize;
     
        
       var a = .5 / lookupSize;
 
-      lookups[ index * 2 + 0 ] = x + a;
-      lookups[ index * 2 + 1 ] = y + a;
+      lookups[ i * 2 + 0 ] = x + a;
+      lookups[ i * 2 + 1 ] = y + a;
       
-      lookups[ index * 2 + 2 ] = x + a;
-      lookups[ index * 2 + 3 ] = y + a;
-
-      lookups[ index * 2 + 4 ] = x + a;
-      lookups[ index * 2 + 5 ] = y + a;
-
-
-    }
   }
 
 
-  var geo = new THREE.BufferGeometry();
+  var geo = new THREE.InstancedBufferGeometry();
 
   var a_position  = new THREE.BufferAttribute( positions , 3 );
   var a_normal    = new THREE.BufferAttribute( normals   , 3 );
-  var a_lookup    = new THREE.BufferAttribute( lookups   , 2 );
+  var a_lookup    = new THREE.InstancedBufferAttribute( lookups, 2, 1, false );
```

Comment:
Magic line...

```THREE.InstancedBufferAttribute( data, attributeSize, numberOfMeshesPerItem, dynamic )```

* ```data``` same as for ```BufferAttribute```
* ```attributeSize``` same as for ```BufferAttribute```
* ```numberOfMeshesPerItem``` **magic** parameter:
 * 0 = buffer is the same as a regular ```BufferAttribute``` but its better to use a regular  ```BufferAttribute```
 * 1 = repeat mesh for each item in the buffer, with this attribute set to that item for the whole mesh (used here)
 * 2 = repeat mesh twice for each item in the buffer...
 * n = etc
* ```dynamic``` is the buffers data dynamicly updated? ```STATIC_DRAW``` vs ```DYNAMIC_DRAW``` with ```.needsUpdate = true``` in this case ```false```

If you only have one instanced attribute then you will likely want to be using 1 as the ```numberOfMeshesPerItem```, however if you have 2 or more then you might want different values.

@toji explains it here: http://blog.tojicode.com/2013/07/webgl-instancing-with.html